app/main.py

```diff
@@ -1,6 +1,19 @@
 from typing import Callable
+from functools import wraps
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    cache_dict = {}
+
+    @wraps(func)
+    def wrapper(*args, **kwargs) -> Callable:
+        key = (args, tuple(kwargs.items()))
+
+        if key in cache_dict:
+            print("Getting from cache")
+            return cache_dict[key]
+
+        print("Calculating new result")
+        cache_dict[key] = func(*args, **kwargs)
+        return cache_dict[key]
```

Comment:
you still have 1 return statements with the same returned value. so you can add `if else` statements but use onle one `return` statement

```suggestion

        if key in cache_dict:
            print("Getting from cache")
        else:
            print("Calculating new result")
            cache_dict[key] = func(*args, **kwargs)
        return cache_dict[key]
```